pennylane/labs/resource_estimation/templates/subroutines.py

```diff
@@ -839,3 +1231,312 @@ def resource_rep(cls, cmpr_ops, num_ctrl_wires) -> CompressedResourceOp:
+        }
+
+    @classmethod
+    def resource_rep(
+        cls, num_bitstrings, num_bit_flips, num_control_wires, num_work_wires, size_bitstring, clean
+    ) -> CompressedResourceOp:  # pylint: disable=too-many-arguments
+        r"""Returns a compressed representation containing only the parameters of
+        the Operator that are needed to compute a resource estimation.
+
+        Args:
+            num_bitstrings (int): the number of bitstrings that are to be encoded
+            num_bit_flips (int): the number of bit flips needed for the list of bitstrings
+            num_control_wires (int): the number of control wires where in the indexes are specified
+            num_work_wires (int): the number of auxiliary wires used for QROM computation
+            size_bitstring (int): the length of each bitstring
+            clean (bool): if True, the work wires are not altered by the QROM operator
+
+        Returns:
+            CompressedResourceOp: the operator in a compressed representation
+        """
+        params = {
+            "num_bitstrings": num_bitstrings,
+            "num_bit_flips": num_bit_flips,
+            "num_control_wires": num_control_wires,
+            "num_work_wires": num_work_wires,
+            "size_bitstring": size_bitstring,
+            "clean": clean,
+        }
+        return CompressedResourceOp(cls, params)
+
+
+class ResourceAmplitudeAmplification(qml.AmplitudeAmplification, ResourceOperator):
+    r"""Resource class for the AmplitudeAmplification template.
+
+    Args:
+        U (Operator): the operator that prepares the state :math:`|\Psi\rangle`
+        O (Operator): the oracle that flips the sign of the state :math:`|\phi\rangle` and does nothing to the state :math:`|\phi^{\perp}\rangle`
+        iters (int): the number of iterations of the amplitude amplification subroutine, default is ``1``
+        fixed_point (bool): whether to use the fixed-point amplitude amplification algorithm, default is ``False``
+        work_wire (int): the auxiliary wire to use for the fixed-point amplitude amplification algorithm, default is ``None``
+        reflection_wires (Wires): the wires to reflect on, default is the wires of ``U``
+        p_min (int): the lower bound for the probability of success in fixed-point amplitude amplification, default is ``0.9``
+
+    Resource Parameters:
+        * U_op (Operator): the operator that prepares the state :math:`|\Psi\rangle`
+        * U_params (dict): the parameters for the U operator
+        * O_op (Operator): the oracle that flips the sign of the state :math:`|\phi\rangle` and does nothing to the state :math:`|\phi^{\perp}\rangle`
+        * O_params (dict): the parameters for the O operator
+        * iters (int): the number of iterations of the amplitude amplification subroutine
+        * num_ref_wires (int): the number of wires used for the reflection
+        * fixed_point (bool): whether to use the fixed-point amplitude amplification algorithm
```

Comment:
Need to have type since we are passing a reference to the class not an instance of the class. 
```suggestion
        * U_op (Type[~.ResourceOperator]): the class of the operator that prepares the state :math:`|\Psi\rangle`
        * U_params (dict): the parameters for the U operator
        * O_op (Type[~.ResourceOperator]): the class of the oracle that flips the sign of the state :math:`|\phi\rangle` and does nothing to the state :math:`|\phi^{\perp}\rangle`
        * O_params (dict): the parameters for the O operator
        * iters (int): the number of iterations of the amplitude amplification subroutine
        * num_ref_wires (int): the number of wires used for the reflection
        * fixed_point (bool): whether to use the fixed-point amplitude amplification algorithm
```